app.js

```diff
@@ -47,6 +53,27 @@ var app_manager = require('./routes/app/router-app-manager'),
     user = require('./routes/user/router-user'),
     farmer = require('./routes/resources/farmer/router-farmer');
 
+var roleMiddleware = function (req, res, next) {//this function gets the role name of the application whose access token was used in the API request
```

Comment:
@nick22891 We have to do some work on this one.  Token Authorization must be handled using another passport framework - and not a custom written middleware.  Checkout [this](https://www.npmjs.com/package/passport-token-auth).